Uploading the initial program that contains -

halelitzhaki opened the pull request:
1. Gets sudo password from user in gui
2. Popups window when new usb device is connected
3. Gets user's choice when selected from the popup window (allow or block) and execute according to it.

halelitzhaki committed:
Uploading the initial program that contains -
1. Gets sudo password from user in gui
2. Popups window when new usb device is connected
3. Gets user's choice when selected from the popup window (allow or block) and execute according to it.

HorizonRL commented on gui/main_window.py:
empty file

HorizonRL commented on .idea/.gitignore:
all of the .idea files should be ignored and not uploaded to git

HorizonRL commented on gui/get_sudo_password_popup.py:
```diff
@@ -0,0 +1,18 @@
+import subprocess
+
+def get_sudo_password_gui():
```
use type hinting

HorizonRL commented on data/usb_device_logs.json:
```diff
@@ -0,0 +1,9 @@
+        "vendor_id": "0781",
+        "product_id": "5567",
+        "serial": "4C530001300524104314",
+        "device": "/dev/bus/usb/003/022",
```
r u sure this isnt port specific?

halelitzhaki committed:
Uploading the initial program that contains -
1. Gets sudo password from user in gui
2. Popups window when new usb device is connected
3. Gets user's choice when selected from the popup window (allow or block) and execute according to it
4. Auto mounting usb device if allowed
5. Adding ML to study if same vendor is being allowed several times to auto-allowing

HorizonRL commented on core/device_manager.py:
```diff
@@ -1,69 +1,160 @@
+from gui.usb_alert import show_usb_alert
+from ml_model.model import predict
+
+LOG_FILE = 'data/usb_device_logs.json'
```
put this inside a logger module of urself (encapsulation, the device_manager wanna log, it dosent care where or how)

HorizonRL commented on core/device_manager.py:
```diff
@@ -1,69 +1,160 @@
+    DEBOUNCE_TIME = 1
+
+    def __init__(self, sudo_password):
+        self.sudo_password = sudo_password
```
if i was writing this, i would design this as follows:
the sudo password will be inside an object called `RootProcessLauncher`,
(`RootProcessLauncher` will get the paswd in the ctor and it will be a member of `USBDeviceManager`)
`RootProcessLauncher` will have a public function called: `execute(command: str)`
this way, u are creation seperation between the knowledge of how to execute root-process and the what (the command itself)

HorizonRL replied:
sudo -S repeats so its an ez addon to this class

HorizonRL commented on core/device_manager.py:
```diff
@@ -1,69 +1,160 @@
+                .communicate(input=f'{self.sudo_password}'.encode())
+            print(f"USB Device: {device.vendor_id} - {device.product_id} mounted successfully at {mount_point}")
+        except subprocess.CalledProcessError as e:
+            print(f"Failed to mount the USB device: {e}")
```
this should be inside another module and function (e.g mounter.mount or something like this)
what if the code fails after the mount (hint: ull have unwanted leftovers) i would recommend to read about `@contextmanager`
its like RAII (from c++ if u know)

HorizonRL commented on core/device_manager.py:
```diff
@@ -1,69 +1,160 @@
+
+    def allow_usb_device(self, device):
+        """ Logic to allow and automatically mount the USB device. """
+        print(f"Allowing and mounting USB Device: {device.vendor_id} - {device.product_id}")
```
this print repeats, create a function for this :-)

HorizonRL commented on core/device_manager.py:
```diff
@@ -1,69 +1,160 @@
+            print(f"USB Device: {device.vendor_id} - {device.product_id} blocked successfully")
+
+        except subprocess.CalledProcessError as e:
+            print(f"Failed to block the USB device: {e}")
```
u r muting the exception, its ok to catch it and log it or something, but ur function failed and now the code will just try to continue like nothing (like ross from rachel even though they where on a break)

HorizonRL commented on core/device_manager.py:
```diff
@@ -1,69 +1,160 @@
+            print(f"Failed to block the USB device: {e}")
+
+    def monitor_usb_devices(self):
+        """ Monitor USB devices and handle them based on user input and model predictions. """
```
way too long funciton

HorizonRL replied:
think about the steps this function is doing, and break a part each step to its logical seperation (either by class, func or module)

HorizonRL commented on core/device_utils.py:
```diff
@@ -46,41 +23,48 @@ def get_block_device(device_node):
-        # Pass the sudo password and run the echo command to append to the file
-        subprocess.Popen(echo_command, stdin=subprocess.PIPE, stderr=subprocess.DEVNULL)\
-            .communicate(input=f'{sudo_password}\n{blocking_command}\n'.encode())
+def log_usb_device(device_info, decision):
```
this ezily could be a class that logs the usb devices :-)

HorizonRL commented on core/device_utils.py:
```diff
@@ -46,41 +23,48 @@ def get_block_device(device_node):
-    except subprocess.CalledProcessError as e:
-        print(f"Failed to block the USB device: {e}")
+
+def disable_auto_mount():
```
i would create a `enable_auto_mount` so when ull stop running (either by bad or happy flow) this will make ur program much more easier to debug and use

HorizonRL commented on core/usb_device.py:
```diff
@@ -0,0 +1,10 @@
+class USBDevice:
```
`@dataclass`

HorizonRL commented on core/usb_device.py:
```diff
@@ -0,0 +1,10 @@
+        self.device_node = device_node
+
+    def __repr__(self):
+        return f"USBDevice(vendor_id={self.vendor_id}, product_id={self.product_id}, serial={self.serial}, device_node={self.device_node})"
```
`__dict__` of `dataclass`

HorizonRL commented on data/usb_device_logs.json:
idk if u should upload this to git

HorizonRL commented on data/vendor_allow_counts.json:
idk if u should upload this to git

HorizonRL commented on gui/usb_alert.py:
has a lot in common with `get_sudo_password`

HorizonRL reviewed (commented):
1. use type hinting

halelitzhaki committed:
Uploading the initial program that contains -
1. Gets sudo password from user in gui
2. Popups window when new usb device is connected
3. Gets user's choice when selected from the popup window (allow or block) and execute according to it
4. Auto mounting usb device if allowed
5. Adding ML to study if same vendor is being allowed several times to auto-allowing

halelitzhaki committed:
Uploading the initial program that contains -
1. Gets sudo password from user in gui
2. Popups window when new usb device is connected
3. Gets user's choice when selected from the popup window (allow or block) and execute according to it
4. Auto mounting usb device if allowed
5. Adding ML to study if same vendor is being allowed several times to auto-allowing

halelitzhaki committed:
Delete ml_model/saved_model.pkl

halelitzhaki committed:
Delete tests directory
